### PR TITLE
feat: added id field to operation schema

### PR DESCRIPTION
**Pull request by gpuente**
## Description:

- added id field to operation schema

### CODE REVIEW

File: src/document/reducer.ts [hunk 1]
```diff
@@ -118,11 +120,15 @@ function updateOperations<T extends Document>(
         }
 
         nextIndex = action.index;
+        operationId = action.id;
+    } else {
+        operationId = uuid();
```
Comment (acaldas) [inline src/document/reducer.ts:125]: This makes the reducer non pure which might be a problem but let's leave it like this for now to make the transition easier.

File: src/document/types.ts [hunk 2]
```diff
@@ -132,6 +132,8 @@ export type Operation<A extends Action = Action> = A & {
     skip: number;
     /** Error message for a failed action */
     error?: string;
+    /** Unique operation id */
+    id?: string;
```
Comment (acaldas) [inline src/document/types.ts:136]: Can we make it required? Since we are assigning an id in the reducer if not provided

Comment (gpuente) [reply src/document/types.ts:136]: I'm keeping this optional so i's compatible with old operations that have not an ID

Comment (acaldas) [reply src/document/types.ts:136]: Ok, let's keep it and refactor later

**Review by acaldas — APPROVED** [review]

**Pull request merged**